internal/values/proto.go

```diff
@@ -84,6 +84,33 @@ func GetTimeValue(in *structpb.Struct, k string) (time.Time, error) {
 	return t, nil
 }
 
+// GetMapString returns a map[string]string value and no error if the given key
+// is found in the provided proto struct input. An error is returned if the key
+// is not found or the value type is not map[string]string.
+func GetMapString(in *structpb.Struct, k string, required bool) (map[string]string, error) {
```

Comment:
Not sure exactly why but I do not love this helper function. Perhaps because it is only used once or the name is not technically correct `GetMapStringString`. I think it is okay to leave for now as I don't really have a suggestion other than just directly doing the conversion in `GetCredentialAttributes`

Reply:
I agree with you. I think my original thought process was that this might be a useful helper function for future attributes that might be needed, such as resource tags.